### PR TITLE
Resource 220 lazy load aws iam users

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [X] All Unit Tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: inspec.yml [hunk 1]
```diff
@@ -6,4 +6,4 @@ copyright: spaterson@chef.io,russell.seymour@turtlesystems.co.uk
 copyright_email: spaterson@chef.io,russell.seymour@turtlesystems.co.uk
 version: 1.81.19
 license: Apache-2.0
-inspec_version: '>= 4.18.108'
+inspec_version: '>= 4.56.17'
```
Comment (clintoncwolfe) [inline inspec.yml:9]: Consider adding a comment explaining why this version is needed

File: libraries/aws_iam_users.rb [hunk 2]
```diff
@@ -2,102 +2,97 @@
...
-                          password_last_used_days_ago: password_last_used_days_ago,
-                          has_console_password:  has_password?(username) }]
+    catch_aws_errors do
+      @aws.iam_client.list_users.flat_map do |response|
+        response.users.each_with_object({}) do |user, hash|
+          hash[:username] = user.arn.split('/').last
+          hash[:user_arn] = user.arn
+          hash[:user_id] = user.user_id
+          hash[:password_ever_used?] = user.password_last_used.present?
+          hash[:password_last_used_days_ago] = ((Time.current - user.password_last_used) / (24*60*60)).to_i
```
[the remark this answers is not in the document]
Comment (sathish-progress) [reply libraries/aws_iam_users.rb:52]: yeah, actually !! will add a check

File: inspec.yml (hunk 1 above)
Comment (sathish-progress) [reply inspec.yml:10]: done

File: libraries/aws_iam_users.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (sathish-progress) [reply libraries/aws_iam_users.rb]: done

**Review by clintoncwolfe — APPROVED** [review]
Looks good, thanks!

**Review by soumyo13 — APPROVED** [review]

**Pull request merged**